src/olympia/addons/serializers.py

```diff
@@ -477,6 +482,18 @@ def _update_admin_review_flag_and_logging(self, version):
             )
 
     def create(self, validated_data):
+        has_source = 'source' in validated_data and validated_data['source']
+        slug = self.addon.slug if self.addon else validated_data.get('addon').slug
+
+        if has_source:
+            timer = StopWatch('addons.serializers.developer_version_serializer.create.')
+            timer.start()
+            log.info(
+                'create, source upload received, addon.slug: %s',
+                slug,
+            )
+            timer.log_interval('1.source_received')
```

Comment:
I suspect this is too late to capture the timing for the source code upload - it will already have been validated (including checking the file validity) by the point `create` is called so a bunch of the work will have been done.

Reply:
Ok, so do we need to have the logging in the view?

Reply:
Yeah, this should probably be done in the view so that you can start the timer before calling `serializer.is_valid()`, then log an internal when you call `.save()`...

The view code is currently the [generic DRF implementation](https://github.com/encode/django-rest-framework/blob/master/rest_framework/mixins.py) though (`CreateModelMixin` and `UpdateModelMixin`), so you'd need to copy that over in our viewset... (And like in devhub, you'd want to check for `source` in `request.data`, before validation)

Reply:
Thanks Mat. I thought it was using the generic DRF implementation, so I guess we need to override that for both create and update, unless there are insertion points where we can add logging. I can check for that but I'm guessing there are not.

Reply:
I moved the logging code into the view.